File: CS/ViewerForm1.cs
using DevExpress.DashboardCommon;
using DevExpress.XtraEditors;
using System;
using System.Threading.Tasks;

namespace DataCounterExample
{
    public partial class ViewerForm1 : XtraForm {
        public ViewerForm1() {
            InitializeComponent();
            dashboardViewer1.Initialized += OnDashboardViewerInitialized;
            dashboardViewer1.MasterFilterSet += OnDashboardViewerMasterFilterSet;
            dashboardViewer1.MasterFilterCleared += OnDashboardViewerMasterFilterCleared;
        }

        async void OnDashboardViewerInitialized(object sender, EventArgs e) {
            simpleLabelItem1.Text = await GetLabelText("choroplethMapDashboardItem1", "Map");
            simpleLabelItem2.Text = await GetLabelText("gridDashboardItem1", "Grid");
        }

        async void OnDashboardViewerMasterFilterSet(object sender, MasterFilterSetEventArgs e) {
            simpleLabelItem2.Text = await GetLabelText("gridDashboardItem1", "Grid");
        }

        async void OnDashboardViewerMasterFilterCleared(object sender, MasterFilterClearedEventArgs e) {
            simpleLabelItem2.Text = await GetLabelText("gridDashboardItem1", "Grid");
        }

        async Task<string> GetLabelText(string itemName, string itemCaption) {
            var data = await dashboardViewer1.GetItemDataAsync(itemName);
            int count = data.GetAxisPoints(data.GetAxisNames()[0]).Count;
            return string.Format("{0}: {1}", itemCaption, count);
        }
    }
}

File: VB/ViewerForm1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.XtraEditors
Imports System
Imports System.Threading.Tasks

Namespace DataCounterExample
	Partial Public Class ViewerForm1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			AddHandler dashboardViewer1.Initialized, AddressOf OnDashboardViewerInitialized
			AddHandler dashboardViewer1.MasterFilterSet, AddressOf OnDashboardViewerMasterFilterSet
			AddHandler dashboardViewer1.MasterFilterCleared, AddressOf OnDashboardViewerMasterFilterCleared
		End Sub

		Private Async Sub OnDashboardViewerInitialized(ByVal sender As Object, ByVal e As EventArgs)
			simpleLabelItem1.Text = Await GetLabelText("choroplethMapDashboardItem1", "Map")
			simpleLabelItem2.Text = Await GetLabelText("gridDashboardItem1", "Grid")
		End Sub

		Private Async Sub OnDashboardViewerMasterFilterSet(ByVal sender As Object, ByVal e As MasterFilterSetEventArgs)
			simpleLabelItem2.Text = Await GetLabelText("gridDashboardItem1", "Grid")
		End Sub

		Private Async Sub OnDashboardViewerMasterFilterCleared(ByVal sender As Object, ByVal e As MasterFilterClearedEventArgs)
			simpleLabelItem2.Text = Await GetLabelText("gridDashboardItem1", "Grid")
		End Sub

		Private Async Function GetLabelText(ByVal itemName As String, ByVal itemCaption As String) As Task(Of String)
			Dim data = Await dashboardViewer1.GetItemDataAsync(itemName)
			Dim count As Integer = data.GetAxisPoints(data.GetAxisNames()(0)).Count
			Return String.Format("{0}: {1}", itemCaption, count)
		End Function
	End Class
End Namespace